src/SignalR/common/Shared/MessageBuffer.cs

```diff
@@ -244,29 +226,27 @@ internal void ResetSequence(SequenceMessage sequenceMessage)
         _currentReceivingSequenceId = sequenceMessage.SequenceId;
     }
 
-    internal void Resend()
+    internal Task Resend(PipeWriter writer)
```

Comment:
Nit:

```suggestion
    internal async Task ResendAsync(PipeWriter writer)
```

And then delete `DoResendAsync` and just have this method body continue. That part is almost simple enough to suggest in a diff, but it doesn't work since it spans unchanged lines.